book/visual-effects.md

```diff
@@ -985,97 +994,26 @@ of stacking contexts is very related to the tree of surfaces.
 [stacking-context]: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context
 [z-index]: https://developer.mozilla.org/en-US/docs/Web/CSS/z-index
 
-To match this use pattern, in Skia, surfaces form a stack. You can
-push a new surface on the stack, raster things to it, and then pop it
-off by blending it with surface below. When traversing the tree of
-stacking contexts, you push a new surface onto the stack every time
-you recurse into a new stacking context, and pop-and-blend every time
-you return from a child stacking context to its parent.
- 
-In real browsers, stacking contexts are formed by HTML elements with
-certain styles, up to any descendants that themselves have such
-styles. The full definition is actually quite complicated, so in this
-chapter we'll simplify by treating every layout object as a stacking
-context.
-
 ::: {.further}
-Mostly, elements [form a stacking context][stacking-context] because
-of CSS properties that have something to do with layering (like
-`z-index`) or visual effects (like `mix-blend-mode`). On the other
-hand, the `overflow` property, which can make an element scrollable,
-does not induce a stacking context, which I think was a
-mistake.[^also-containing-block] The reason is that inside a modern
-browser, scrolling is done on the GPU by offsetting two surfaces.
-Without a stacking context the browser might (depending on the web
-page structure) have to move around multiple independent surfaces with
-complex paint orders, in lockstep, to achieve scrolling. Fixed- and
-sticky-positioned elements also form stacking contexts because of
-their interaction with scrolling.
+[This blog post](https://ciechanow.ski/alpha-compositing/) gives a really nice
+visual overview of many of the same concepts explored in this chapter,
+plus way more content about how a library such as Skia might implement features
+like raster sampling of vector graphics for lines and text, and interpolation
+of surfaces when their pixel arrays don't match resolution or orientation. I
+highly recommend it.
 :::
 
-[^also-containing-block]: While we're at it, perhaps scrollable
-elements should also be a [containing block][containing-block] for
-descendants. Otherwise, a scrollable element can have non-scrolling
-children via properties like `position`. This situation is very
-complicated to handle in real browsers.
-
-[containing-block]: https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block
-
-Opacity and alpha
-=================
-
-[canvas]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/canvas
-[webgl]: https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API
-[webassembly]: https://developer.mozilla.org/en-US/docs/WebAssembly
-[canvaskit]: https://skia.org/docs/user/modules/canvaskit/
-
-Color mixing happens when multiple page elements overlap. The easiest
```

Comment:
I guess your new text is intended to fold in this content also, but with fewer words?

Reply:
Yeah I was hoping to fold in theory sections. I think this somewhat worked but I could have got there with less rewriting. The biggest improvement was adding RGBA color to contrast with opacity.